apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.UpdateObject.cs

```diff
@@ -41,6 +41,19 @@ private ObjectsResource.UpdateRequest CreateUpdateObjectRequest(Object obj, Upda
             GaxPreconditions.CheckArgument(obj.Bucket != null, nameof(obj), "The Bucket property of the object to update is null");
             GaxPreconditions.CheckArgument(obj.Name != null, nameof(obj), "The Name property of the object to update is null");
             var request = Service.Objects.Update(obj, obj.Bucket, obj.Name);
+
+            if (options?.RetentionMode != null || options?.RetainUntilTime != null)
+            {
+                obj.Retention = new Object.RetentionData
+                {
+                    Mode = options.RetentionMode,
+                    RetainUntilTimeDateTimeOffset = options.RetainUntilTime
+                };
+            }
+            else
+            {
+                obj.Retention = null;
```

Comment:
Do we need to set null explicitly?

Reply:
Yes, we need to. This is for the use case where we need to remove retention from an object. So in that case fetched bucket object will already have a retention configure on it, which we need to mark as null and update.

Reply:
Okay, on thinking more on it, there could be also a use case where user is updating an object and at the same time wants existing retention of the object to be same as it is( not removed ). In that case they may not provide  RetentionMode and RetainUntilTime.
To Handle such cases as well do you suggest that we do not mark existing retention property of fetched object null here. Introduce a new bool property say options.RemoveRetention and mark retention of existing object as null only if it is set to true ? What do you think ?